src/tiled/commandbutton.cpp

```diff
@@ -42,7 +43,11 @@ CommandButton::CommandButton(QWidget *parent)
     setPopupMode(QToolButton::MenuButtonPopup);
     setMenu(mMenu);
 
-    connect(mMenu, SIGNAL(aboutToShow()), SLOT(populateMenu()));
+    // Populate the menu once in order to register shortcuts beforehand
+    CommandManager::instance()->populateMenu(mMenu);
+
+    connect(mMenu, &QMenu::aboutToShow, 
+            CommandManager::instance(), [this]() { CommandManager::instance()->populateMenu(mMenu); });
```

Comment:
Since we know the `CommandButton` will not outlive the `CommandManager`, we can simplify this to:
```c++
    connect(mMenu, &QMenu::aboutToShow, 
            [this]() { CommandManager::instance()->populateMenu(mMenu); });
```
Same for the connection to `mUi->menuCommand` in the `mainwindow.cpp`.